src/trfs_fast/generation.py

```diff
@@ -116,7 +116,7 @@ def generate_minimal(
         batch_size, context_length = input_ids.shape
         cache_length = cache_length if cache_length is not None else max_new_tokens
 
-        model_kwargs["valid_past_index"] = 0
+        model_kwargs["valid_past_index"] = torch.tensor(0).to(input_ids.device)
```

Comment:
@gante I'm not sure if it's a good thing to have a slicing index on device, slice aten::slice is a cpu op?